### ISSUE TITLE
invalid column number (-1) returned by merlin/outline

### ISSUE THREAD

**Issue by tddsg** [ocaml/merlin#638]
Hi there,

It seems that `merlin` sometimes return an invalid column number (-1), which make the feature `merlin-imenu` not working properly. This happens to this file: https://github.com/facebook/infer/blob/master/infer/src/IR/Localise.ml 

Here is the command of Emacs that I send to get the outline ``(merlin/send-command `outline)``. The output is kind of big, hence I attach it here: [outline.txt](https://github.com/ocaml/merlin/files/960744/outline.txt). You can see the column number `-1` appear quite a number of times. 

Could you double check this issue?
Thanks!

**Comment by let-def** [comment]
I guess it is just the derived function which are not given any position by deriving plugins.

**Comment by tddsg** [comment]
Thank you for the information. The `-1` number appear in this structure:
`((assoc) (start (assoc) (line . 1) (col . -1)) (end (assoc) (line . 1) (col . -1)) (name . compare_t) (kind . Value) (children))`

I searched in the source code but didn't see any text `compare_t`. How can `merlin` find it?

There are several relevant texts like: `[@@deriving compare]`, `[%compare.equal : string list]`, `let equal = [%compare.equal : t]`.

Is `compare_t` from one of them?

**Comment by let-def** [comment]
Yes. `ppx_deriving` is explained here: https://github.com/whitequark/ppx_deriving

ppx is a feature which allows external programs to generate OCaml code.
*deriving* is one of these programs which will generate comparison, printing, etc, functions from the a type declaration.
These programs are free to give a location or not to the generated code. `line 1 col -1` is the convention of OCaml compiler for `no location`.

**Comment by tddsg** [comment]
Thanks for the update! In this case, is it feasible to return the location to one of the lines using `ppx_deriving`? To me, this makes more sense than returning `-1`.

**Comment by let-def** [comment]
Merlin cannot do anything about that. This is up to the implementors of ppx.
This even looks like a bug that might be worth reporting (check which ppx is being applied before reporting :))
